Userland/Libraries/LibGfx/ImageFormats/ImageDecoder.cpp

```diff
@@ -88,17 +86,17 @@ static OwnPtr<ImageDecoderPlugin> probe_and_sniff_for_appropriate_plugin_with_kn
     return {};
 }
 
-RefPtr<ImageDecoder> ImageDecoder::try_create_for_raw_bytes(ReadonlyBytes bytes, Optional<ByteString> mime_type)
+ErrorOr<RefPtr<ImageDecoder>> ImageDecoder::try_create_for_raw_bytes(ReadonlyBytes bytes, Optional<ByteString> mime_type)
```

Comment:
It feels a *little* awkward that the result is either:
* Error, because we choked while decode
* Null, because we couldn't find a decoder
* Image, happy path

There's clearly existing code depending on this, so I'm just thinking out loud. But I would sort of expect this to be `ErrorOr<NonnullRefPtr<>>`, and return `Error("No decoder")` instead of null :man_shrugging:

Reply:
It's a bit weird! But "we failed to decode this file that we recognize and that we thought we should be able to decode, but maybe the data is broken or our decoder not quite right" is kind of different from "we have no idea what this data is", so it's also a bit strange to return an Error for both maybe? /shrug